src/test/java/bio/terra/common/logging/LoggingTest.java

```diff
@@ -164,7 +164,7 @@ public void testStructuredLogging(CapturedOutput capturedOutput) {
     String event3 = null;
     String event4 = null;
     for (String line : lines) {
-      String message = (String) readJson(line, "$.message");
+      String message = readJson(line, "$.message");
```

Comment:
How does this work without specifying a template parameter value? I'd have expected `readJson<String>()` or something.

Reply:
My guess is that because of target type type inference i.e. the assignment to `String message`, but I won't claim to be able to predict Java's type inference.
https://docs.oracle.com/javase/tutorial/java/generics/genTypeInference.html